src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -152,29 +155,38 @@ private List<IRepository> GetRepos(VssConnection organizationConnection, TeamPro
     /// <returns>A list of repositories the user has access to.</returns>
     public IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IDeveloperId developerId)
     {
-        if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+        return Task.Run(() =>
         {
-            return Task.Run(() =>
+            try
             {
-                var exception = new NotSupportedException("Authenticated user is not the an azure developer id.");
-                return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
-            }).AsAsyncOperation();
-        }
+                if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+                {
+                    var exception = new NotSupportedException("Authenticated user is not the an azure developer id.");
+                    return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
+                }
 
-        _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);
+                _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);

```

Comment:
>_azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId); [](http://example.com/codeflow?start=16&length=67)

(i'm new to c# and ??= and Task.Run - so forgive my questions)

How does _azureHierarchy stay correct.

Say someone calls GetRepos with azureDeveloperId X, and that sets _azureHierarchy.  Then someone calls GetRepositoriesAsync with azureDeveloperId Y.

Who is GetOrganizations [on line 171] being called for - user X or user Y?

Reply:
Hrm - I think i just don't understand line 162

    developerId is not DeveloperId.DeveloperId azureDeveloperId